Upgrading to latest rustc

dsavitskiy opened the pull request.

dsavitskiy commented:
bors try

dsavitskiy commented:
bors try

tiagolobocastro commented on scripts/rust-linter.sh:
```diff
@@ -1,4 +1,26 @@
+    -A clippy::unnecessary-lazy-evaluations \
+    -A clippy::unnecessary-to-owned \
+    -A clippy::unwrap-or-else-default \
+    -A clippy::useless-conversion
```
Are all these allows necessary now? :(

dsavitskiy replied:
These are new lints introduced by the latest clippy. Some of them look quite useful like "await-holding-lock". I'm going to create a separate ticket to fix them.

dsavitskiy commented:
bors try

dsavitskiy commented:
bors try

dsavitskiy commented:
bors try

dsavitskiy commented:
bors try

dsavitskiy commented:
bors merge

dsavitskiy commented:
bors merge

dsavitskiy commented:
bors try

dsavitskiy commented:
bors merge

dsavitskiy commented:
bors merge

dsavitskiy commented:
bors merge

dsavitskiy committed:
chore(ci): updating to latest rustc and nix

Following dependencies updated:
rustc: 1.56.1 -> 1.66 stable
cargo: 1.58.0-nightly -> 1.68.0-nightly (2023-01-04)
rustfmt: 1.4.38-nightly -> 1.5.1-nightly (2023-01-07)
clippy: 0.1.58-nightly -> 0.1.68-nightly (2023-01-07)

nix: pre-21.11 -> 22.11
fio: 3.26 -> 3.33
nvme-cli: pinned to 1.16 (newer 2.x versions break tests)

Updating to the latest Rust required updating to the latest Nix packages
(version 22-11). That required bumping fio version.

Signed-off-by: Dmitry Savitskiy <dmitry.savitskiy@datacore.com>

dsavitskiy commented:
bors merge
